From 61c44e24ea212b92bf6a420b94070ee6fc715811 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 8 May 2019 10:08:31 +0100
Subject: [PATCH] BCM270X_DT: usb: Refactor DTS and overlays

Move the IRQ interrupt declaration in the usb node before the FIQ
declaration, so that the dwc2 driver will find it. Name the
interrupts appropriately so that the dwc_otg driver can still find
them. Then remove the interrupt rewriting from the overlays.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm270x.dtsi                 | 6 ++++--
 arch/arm/boot/dts/overlays/dwc-otg-overlay.dts | 6 ------
 arch/arm/boot/dts/overlays/dwc2-overlay.dts    | 2 --
 3 files changed, 4 insertions(+), 10 deletions(-)

--- a/arch/arm/boot/dts/bcm270x.dtsi
+++ b/arch/arm/boot/dts/bcm270x.dtsi
@@ -131,8 +131,10 @@
 			compatible = "brcm,bcm2708-usb";
 			reg = <0x7e980000 0x10000>,
 			      <0x7e006000 0x1000>;
-			interrupts = <2 0>,
-				     <1 9>;
+			interrupt-names = "usb",
+					  "soft";
+			interrupts = <1 9>,
+				     <2 0>;
 		};
 
 		v3d@7ec00000 { /* vd3 */
--- a/arch/arm/boot/dts/overlays/dwc-otg-overlay.dts
+++ b/arch/arm/boot/dts/overlays/dwc-otg-overlay.dts
@@ -6,14 +6,8 @@
 
 	fragment@0 {
 		target = <&usb>;
-		#address-cells = <1>;
-		#size-cells = <1>;
 		__overlay__ {
 			compatible = "brcm,bcm2708-usb";
-			reg = <0x7e980000 0x10000>,
-			      <0x7e006000 0x1000>;
-			interrupts = <2 0>,
-				     <1 9>;
 			status = "okay";
 		};
 	};
--- a/arch/arm/boot/dts/overlays/dwc2-overlay.dts
+++ b/arch/arm/boot/dts/overlays/dwc2-overlay.dts
@@ -10,8 +10,6 @@
 		#size-cells = <1>;
 		dwc2_usb: __overlay__ {
 			compatible = "brcm,bcm2835-usb";
-			reg = <0x7e980000 0x10000>;
-			interrupts = <1 9>;
 			dr_mode = "otg";
 			g-np-tx-fifo-size = <32>;
 			g-rx-fifo-size = <256>;
